shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/controllers/ManagedBridgeController.java

```diff
@@ -0,0 +1,170 @@
+        //        }
+
+        Secret secret = managedBridgeService.fetchOrCreateBridgeSecret(managedBridge);
+
+        if (secret == null) {
+            LOGGER.info("The Secret for the ManagedBridge with id '{}' has been not created yet.",
+                    managedBridge.getSpec().getId());
+            if (!status.isConditionTypeFalse(ConditionTypeConstants.READY)) {
+                status.markConditionFalse(ConditionTypeConstants.READY);
+            }
+            if (!status.isConditionTypeFalse(ManagedBridgeStatus.SECRET_AVAILABLE)) {
+                status.markConditionFalse(ManagedBridgeStatus.SECRET_AVAILABLE);
+            }
+            return UpdateControl.updateStatus(managedBridge).rescheduleAfter(reconcileIntervalSeconds);
+        } else if (!status.isConditionTypeTrue(ManagedBridgeStatus.SECRET_AVAILABLE)) {
+            LOGGER.info("Secret for ManagedBridge with id '{}' has been created.", managedBridge.getSpec().getId());
+            status.markConditionTrue(ManagedBridgeStatus.SECRET_AVAILABLE);
+        }
+
+        // Nothing to check for ConfigMap
+        ConfigMap configMap = managedBridgeService.fetchOrCreateBridgeConfigMap(managedBridge, secret);
+        if (!status.isConditionTypeTrue(ManagedBridgeStatus.CONFIG_MAP_AVAILABLE)) {
+            status.markConditionTrue(ManagedBridgeStatus.CONFIG_MAP_AVAILABLE);
+        }
+
+        KnativeBroker knativeBroker = managedBridgeService.fetchOrCreateKnativeBroker(managedBridge, configMap);
+        String path = extractBrokerPath(knativeBroker);
+
+        if (path == null) {
+            LOGGER.info("The Knative Broker Resource for ManagedBridge '{}' in namespace '{}' is not ready",
+                    managedBridge.getMetadata().getName(),
+                    managedBridge.getMetadata().getNamespace());
+            if (!status.isConditionTypeFalse(ConditionTypeConstants.READY)) {
+                status.markConditionFalse(ConditionTypeConstants.READY);
+            }
+            if (!status.isConditionTypeFalse(ManagedBridgeStatus.KNATIVE_BROKER_AVAILABLE)) {
+                status.markConditionFalse(ManagedBridgeStatus.KNATIVE_BROKER_AVAILABLE);
+            }
+            return UpdateControl.updateStatus(managedBridge).rescheduleAfter(reconcileIntervalSeconds);
+        } else if (!status.isConditionTypeTrue(ManagedBridgeStatus.KNATIVE_BROKER_AVAILABLE)) {
+            status.markConditionTrue(ManagedBridgeStatus.KNATIVE_BROKER_AVAILABLE);
+            LOGGER.info("The Knative Broker Resource for ManagedBridge '{}' in namespace '{}' is ready", managedBridge.getMetadata().getName(), managedBridge.getMetadata().getNamespace());
+        }
+
+        if (!status.isReady()) {
+            metricsService.onOperationComplete(managedBridge, MetricsOperation.CONTROLLER_RESOURCE_PROVISION);
+            status.markConditionTrue(ConditionTypeConstants.READY);
+            //TODO - notify manager
```

Comment:
Do we have a related JIRA?

Reply:
Yes. I'll update the comment to include the JIRA ticket that will add this functionality.